### PR TITLE
WIP: [Torch] Renamed Torch to TorchCUDA, and added a CPU-only Torch

### PR DESCRIPTION
**Pull request by stemann**
Currently a WIP. Aims at resolving https://github.com/FluxML/Torch.jl/issues/20.

Similar approach as for ONNXRuntime - separate binaries for CPU-only and CUDA etc. (CPU: #4369, CUDA: #4386).

Relates to: #1529

### CODE REVIEW

File: T/Torch/Torch/build_tarballs.jl [hunk 1]
```diff
@@ -54,8 +58,6 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
```
Comment (giordano) [inline T/Torch/Torch/build_tarballs.jl:60]: We'll eventually require `dependencies` to be a vector of `AbstractDependency`s
```suggestion
dependencies = Dependency[
```

**Comment by vchuravy** [comment]
Does this need to be a different package? Or can we turn this into a variant?

**Comment by stemann** [comment]
> Does this need to be a different package? Or can we turn this into a variant?

If eventually both CPU, CUDA and ROCm builds are available, wouldn't the different sets of dependencies necessitate different packages?

File: T/Torch/Torch/build_tarballs.jl (hunk 1 above)
Comment (stemann) [reply T/Torch/Torch/build_tarballs.jl]: Done

**Comment by vchuravy** [comment]
You still have the same issue that if Torch.jl includes `Torch_jll, TorchCUDA_jll,...` all dependencies must be fulfilled. Better to use a platform tag and platform specific dependencies.

**Comment by stemann** [comment]
> You still have the same issue that if Torch.jl includes `Torch_jll, TorchCUDA_jll,...` all dependencies must be fulfilled. Better to use a platform tag and platform specific dependencies.

Right, I see. Is there some discussion/registry of defined/reserved platform tags? I.e. that "cuda" is reserved for CUDA and so on for e.g. "amd_rocm", "intel_oneapi", "apple_coreml", ...

**Comment by stemann** [comment]
Superseded by #4554

**Pull request closed without merging**